src/test/commands.datasource.entry.test.ts

```diff
@@ -0,0 +1,743 @@
+			});
+		});
+
+		afterEach( () => {
+			if (fs.existsSync(f)) {
+				fs.unlinkSync(f);
+			}
+		});
+
+		it('should create a datasource entry inside a datasource when handing over valid parameters', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const oldLen: number = dvConfig.spec.env.length;
+			const eName: string = 'NEWKEY';
+			const eValue: string = 'NEWVALUE';
+			createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, dsConfig, f, eName, eValue)
+				.then( (created) => {
+					if (created) {
+						dvConfig.spec.env.length.should.equal(oldLen+1);
+						should.exist(dvConfig.spec.env.find( (element) => {
+							return element.name === utils.generateFullDataSourceConfigEntryKey(dsConfig, eName) && element.value === eValue;
+						}));
+						done();
+					} else {
+						done(new Error('Execution of the Create DataSource Entry command returned false'));
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should not create a datasource entry inside a datasource when handing over invalid model', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const eName: string = 'NEWKEY';
+			const eValue: string = 'NEWVALUE';
+			createDSEntryCommand.handleDataSourceEntryCreation(undefined, dsConfig, f, eName, eValue)
+				.then( (created) => {
+					if (created) {
+						done(new Error('Execution of the Create DataSource Entry command returned true, but should not.'));
+					} else {
+						done();
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should not create a datasource entry inside a datasource when handing over invalid datasource config', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const eName: string = 'NEWKEY';
+			const eValue: string = 'NEWVALUE';
+			createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, undefined, f, eName, eValue)
+				.then( (created) => {
+					if (created) {
+						done(new Error('Execution of the Create DataSource Entry command returned true, but should not.'));
+					} else {
+						done();
+					}
+				})
```

Comment:
Sonar is mentioning that this inlined function is repeated several times in the file.  Would be better to factorize it.

Reply:
I am not going to change that.

Reply:
why not avoiding duplicate code in the same file?

Reply:
because it will remove readability from the test itself.

Reply:
I find more readable something like:

```
createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, undefined, f, eName, eValue)
	.then(checkDataSourceEntryCreationReturnTrue(done))
	.catch( (err) => {
		done(err);
	});
```
than

```
createDSEntryCommand.handleDataSourceEntryCreation(dvConfig, undefined, f, eName, eValue)
	.then( (created) => {
		if (created) {
			done(new Error('Execution of the Create DataSource Entry command returned true, but should not.'));
		} else {
			done();
		}
	})
```

Reply:
it has the following advantages:
- we can give a clear name to the function
- it avoids duplicated code
- it limits depth level of nested code
- shorter code